src/Workspaces/CoreTestUtilities/ExportProviderCache.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Composition;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Test.Utilities
+{
+    public static class ExportProviderCache
```

Comment:
Can we move the non-cache methods to some other type? It's very confusing as these methods don't have anything to do with caching, which makes consumption confusing later.

Reply:
➡️ This type acts as a cache which facilitates the efficient creation of `ExportProvider` instances, and caches various immutable data objects required for this construction. The name seems reasonable in this context. Normally this response would be 💭 , but given the size of this PR and the marginal benefit of refactoring here I would prefer to leave it.

Reply:
Some methods act as a cache: some methods aren't. I guess it was very confusing from my perspective because some methods are OK to call and some are evil to call in various places.